auth/src/main/java/com/firebase/ui/auth/viewmodel/email/EmailProviderResponseHandler.java

```diff
@@ -54,16 +61,29 @@ public void onSuccess(AuthResult result) {
                     @Override
                     public void onFailure(@NonNull Exception e) {
                         if (e instanceof FirebaseAuthUserCollisionException) {
-                            // Collision with existing user email, it should be very hard for
-                            // the user to even get to this error due to CheckEmailFragment.
-                            ProviderUtils.fetchTopProvider(getAuth(), email)
-                                    .addOnSuccessListener(new StartWelcomeBackFlow(email))
-                                    .addOnFailureListener(new OnFailureListener() {
-                                        @Override
-                                        public void onFailure(@NonNull Exception e) {
-                                            setResult(Resource.<IdpResponse>forFailure(e));
-                                        }
-                                    });
+                            if (AnonymousUpgradeUtils.canUpgradeAnonymous(getAuth(),
```

Comment:
Maybe instead of overloading this branch we should have the `createOrLink` method transform collision errors into a custom error class with a `Continuation`?

Reply:
TBD.